Sort menu provider

peter-budo opened the pull request:
Creating PR for convenient way to keep discussion easier to track and interact

Commit:
Add basic award sorting

Commit:
Now possible to filter awards on category [rough implementation]

Commit:
Extract common functionality into base class

Commit:
Add state keeping

Commit:
Extract menu strings into strings.xml

Commit:
Update changelog

Commit:
Rename method from filterAwards(String) to filterItems(String)

peter-budo committed:
Merge branch 'dev' into award-sorting

peter-budo committed:
• Provided AbstractSorter that will be extended by all sorter classes
 • Added constructor for AwardSorter in favour of static method calls

peter-budo committed:
Merge branch 'dev' into award-sorting

peter-budo committed:
Merge branch 'award-sorting-suggestion' into award-sorting

peter-budo committed:
• Renamed By category to Select category
 • Moved By progress above Select category
 • Added setActionBarSubTitle to BaseFragment
 • Updating action bar subtitle on menus selections

Commit:
Update menu to remove indicator

peter-budo committed:
• Simplified context menus logic

peter-budo committed:
Menu provider
 • Created own MenuProvider to make same experience like with ShareActionProvider
 • Added support comp library
 • Had to change from FrameActivity to ActionBarActivity
 • Theme had to be changed to AppComp

 It does work great
 Given change of theme styles need to be updated to match with dev styling

Commit:
Change theme to AppCompat Light+DarkActionBar

peter-budo committed:
Merge branch 'dev' into sort-menu-provider

Conflicts:
	BF4Intel/src/main/assets/changelog.htm

peter-budo committed:
Changelog for app version 1.0.1

peter-budo committed:
Merge branch 'dev' into sort-menu-provider

Conflicts:
	BF4Intel/src/main/assets/changelog.htm

peter-budo committed:
• Bumped max number of menus before "more options" menu item showed to indicate longer list of menus
 • Split menu on filter and options
 • Updated user LoginActivity with AppCompt theme

Problems:
 • When typing player name text is black instead of holo
 • The actual search does not happen (what connection has it to activity theme???)

peter-budo committed:
Merge branch 'dev' into sort-menu-provider

Commit:
Make SearchView work with AppCompat v7

peter-budo committed:
• Removed YOLOs

peter-budo committed:
Merge branch 'dev' into sort-menu-provider

Conflicts:
	BF4Intel/src/main/assets/changelog.htm

peter-budo committed:
Removed unused import

peter-budo committed:
Fixed ShareActionProvider in NewsArticleFragment

peter-budo committed:
Fixed crash when menu size less then maximum menu count for initial display

peter-budo committed:
Fixed crash if expansion menu title not provided

peter-budo committed:
Filter menu
 • Added filter to action bar
 • Corrected previous resources to detect selection of menu items between sort and filter

peter-budo commented:
@karllindmark there are 6 outstanding "issues"
- [x] With addition of filter icon action bar is crowded and it causes issues with showing sub title. Option would be to move refresh into drop down (show = never) and maybe later implement pull refresh 
- [x] Shared preferences in awards need proper check in case I messed up something when splitting things in two menus
- [x] subtitle need persistence on rotation, that will mean setting subtitle 'onResume()'
- [x] ActionBar "up" caret icon is hamburger when viewing details (should be caret)
- [x] Make the limit 5+ (ie 6 causes the "fold")
- [x] Make medal text "max lines 2" to not overlap value on smaller devices
- [ ] When user select `Sort by categories` or `Sort by progress` reset currently selected index and let it jump at the beggining

Give it a go and let me know. Tonight clan practice so I may not be very responsive

PS: Also fix on navigation drawer spacing

peter-budo committed:
Extracted creation procedure of new MenuProvider

peter-budo committed:
Merge branch 'dev' into sort-menu-provider

peter-budo committed:
Increased max number of items before expandable `show more options` displayed

peter-budo committed:
Changed use of android.app.ActionBar to android.support.v7.ActionBar across application

peter-budo committed:
Sorted burger vs caret war, so now we need caret as expected

peter-budo committed:
Fixed issue where awards/assignments description text overlapping with counter on small size or low density screens

peter-budo committed:
• Changed refresh button spec from always to never as to make space for filter and sort
 • Renamed sort, just dropped 1 from name

peter-budo committed:
Persisted sub-title and display of it on initial view

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/provider/MenuProvider.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    private Context context;
+    private int maximumMenuCount = 5;
```
Should possibly be converted to a constant, as it's not being assigned elsewhere (as far as I can see).

peter-budo replied:
I could not decide if to make it constant or not. To leave it as variable makes it little flexible in case that we decide to extract some stuff to library and re-use it across number of projects. It does not leave you in that awkward situation that if you want to change number of items you will affect all projects that given version of library is used.

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/awards/AwardGridFragment.java:
```diff
@@ -134,4 +147,59 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+    }
+
+    @Override
+    public void onMenuSelected(MenuItem item) {
```
Where do we declare the itemIds for the different menu alternatives? A bit error prone using the magic numbers.

peter-budo replied:
This is little tricky. It does really begs that `Show all` and `Show progress` should be inherited from parents, which is nice as it it declared once and all children get access to it. 
However if we do that and extract base classes to library project then other future applications  will inherit it no mutter if they need/use 
Nevertheless I will in mean time make it string comperable

peter-budo committed:
Small tweaking making menu selection less error prone. However the sort menus begs for map introduction

peter-budo committed:
• Removed empty lines
 • Corrected SharedPreferences editing
 • Double line layout on awards details for smaller density devices

Commit:
Make sure soldier box is visibility is being toggled correctly

Commit:
Update navigation drawer indices

Commit:
Add filter category to the shared preferences
